scylla/tests/ccm-integration/logged_cli.rs

```diff
@@ -0,0 +1,344 @@
+        let stderr_task = Self::stream_reader(
+            child.stderr.take().expect("Failed to capture stderr"),
+            self.file.as_ref().unwrap().clone(),
+            format!("{:15} -> ", format!("stderr[{}]", run_id)),
+            Some(&mut stderr),
+        );
+
+        let (_, _, status) = tokio::join!(stdout_task, stderr_task, child.wait());
+        match status {
+            Ok(status) => {
+                match status.code() {
+                    Some(code) => {
+                        writer
+                            .lock()
+                            .await
+                            .write_all(
+                                format!(
+                                    "{:15} -> status = {}\n",
+                                    format!("exited[{}]", run_id),
+                                    code
+                                )
+                                .as_bytes(),
+                            )
+                            .await
+                            .ok();
+                    }
+                    None => {
+                        writer
+                            .lock()
+                            .await
+                            .write_all(
+                                format!(
+                                    "{:15} -> status = unknown\n",
+                                    format!("exited[{}]", run_id)
+                                )
+                                .as_bytes(),
+                            )
+                            .await
+                            .ok();
+                    }
+                }
+                if !allow_failure && !status.success() {
+                    let tmp = stderr.deref();
+                    return Err(Error::msg(format!(
+                        "Command `{} {}` failed: {}, stderr: \n{}",
+                        command,
+                        args.join(" "),
+                        status.to_string(),
+                        std::str::from_utf8(tmp)?
+                    )));
+                }
+                Ok(status)
+            }
+            Err(e) => {
+                writer
+                    .lock()
+                    .await
+                    .write_all(
+                        format!(
+                            "{:15} -> failed to wait on child process: = {}\n",
+                            format!("exited[{}]", run_id),
+                            e
+                        )
+                        .as_bytes(),
+                    )
+                    .await
+                    .ok();
+                Err(Error::from(e).context(format!(
+                    "Command `{} {}` failed",
+                    command,
+                    args.join("")
+                )))
+            }
+        }
+    }
+
+    async fn stream_reader<T>(
+        stream: T,
+        writer: Arc<Mutex<File>>,
+        prefix: String,
+        mut buffer: Option<&mut Vec<u8>>,
```

Comment:
```suggestion
        buffer: Option<&mut Vec<u8>>,
```

Reply:
Try experimenting with matching on `buffer.as_ref()` or `buffer.as_mut()` instead of matching on plain `buffer`. Might solve the issue.

Reply:
`as_mut` requires `buffer` to be mut, `as_ref` goes wrong way, but `match buffer` worked well, take a look